samples/sample_decode/src/pipeline_decode.cpp

```diff
@@ -573,6 +573,35 @@ mfxStatus CDecodingPipeline::Init(sInputParams *pParams)
         if (m_diMode)
             m_mfxVppVideoParams.vpp.Out.PicStruct = MFX_PICSTRUCT_PROGRESSIVE;
 
+        if (pParams->bPowerSavingMode)
+        {
+            // Initialize extended buffer for frame processing
+            // - FF scaling        VPP scaling filter
+            // - mfxExtVPPDoUse:   Define the processing algorithm to be used
+            // - mfxExtVPPScaling: Scaling configuration
+            // - mfxExtBuffer:     Add extended buffers to VPP parameter configuration
+            mfxExtVPPDoUse extDoUse;
+            memset(&extDoUse, 0, sizeof(extDoUse));
+            mfxU32 tabDoUseAlg[1];
+            extDoUse.Header.BufferId = MFX_EXTBUFF_VPP_DOUSE;
+            extDoUse.Header.BufferSz = sizeof(mfxExtVPPDoUse);
+            extDoUse.NumAlg = 1;
+            extDoUse.AlgList = tabDoUseAlg;
+            tabDoUseAlg[0] = MFX_EXTBUFF_VPP_DENOISE;
+
+            mfxExtVPPScaling scalingConfig;
+            memset(&scalingConfig, 0, sizeof(scalingConfig));
+            scalingConfig.Header.BufferId = MFX_EXTBUFF_VPP_SCALING;
+            scalingConfig.Header.BufferSz = sizeof(mfxExtVPPScaling);
+            scalingConfig.ScalingMode = MFX_SCALING_MODE_LOWPOWER;        // low power mode
```

Comment:
It's weird to see mfxExtVPPScaling for CSC case (this patch is motivated by internal thread where CSC is discussed). If mfxExtVPPScaling  can turns on VEBOX+SFC for CSC, it requires some clarification in the spec.

Reply:
Sorry, denoise is a typo. I changed that.

Reply:
ok. Thanks, I will try to do some clarification in the spec today or early next week. Will go home now. Have a good day @dmitryermilov

Reply:
Thanks:) You too!

Reply:
I've added some clarification in document and code. Please review.